core/common/src/main/kotlin/eu/kanade/tachiyomi/util/storage/DiskUtil.kt

```diff
@@ -102,26 +105,85 @@ object DiskUtil {
     }
 
     /**
-     * Mutate the given filename to make it valid for a FAT filesystem,
-     * replacing any invalid characters with "_". This method doesn't allow hidden files (starting
-     * with a dot), but you can manually add it later.
+     * Transform a filename fragment to make it safe to use on almost
+     * all commonly used filesystems. You can pass an entire filename,
+     * or just part of one, in case you want a specific part of a long
+     * filename to be truncated, rather than the end of it.
+     *
+     * Characters that are potentially unsafe for some filesystems are
+     * replaced with underscores. This includes the standard ones from
+     * https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+     * but does allow any other valid Unicode code point.
+     *
+     * Excessively long filenames are truncated, by default to 240
+     * bytes. Note that the truncation is based on bytes rather than
+     * characters (code points), because this is what is relevant to
+     * filesystem restrictions in most cases.
+     *
+     * Leading periods are stripped, to avoid the creation of hidden
+     * files by default. If a hidden file is desired, a period can be
+     * prepended to the return value from this function.
+     *
+     * If the optional argument disallowNonEnglish is set to true,
+     * then ALL non-English characters and ANYTHING outside the ASCII
+     * range are replaced not with underscores, but with their
+     * hexadecimal encoding. This is to make it so that distinct
+     * non-English titles of things remain distinct, since not all
+     * places where this function is used also take care of
+     * disambiguation.
+     *
+     * We could instead replace only non-English characters known to
+     * be problematic, but so far nobody with a non-Unicode-compliant
+     * device has been able to provide either directions to reproduce
+     * their issue nor any documentation or tests that would allow us
+     * to determine which characters are problems and which are not.
      */
-    fun buildValidFilename(origName: String): String {
+    fun buildValidFilename(
+        origName: String,
+        maxBytes: Int = MAX_FILE_NAME_BYTES,
+        disallowNonEnglish: Boolean = false,
+    ): String {
         val name = origName.trim('.', ' ')
         if (name.isEmpty()) {
             return "(invalid)"
         }
         val sb = StringBuilder(name.length)
         name.forEach { c ->
-            if (isValidFatFilenameChar(c)) {
+            if (disallowNonEnglish && c >= 0x80.toChar()) {
```

Comment:
```suggestion
            if (disallowNonEnglish && c >= 0x7F.toChar()) {
```
0x7F is the DEL control character which is probably not a good fit for file names even if it's within the original ASCII list 😄 

cf. https://en.wikipedia.org/wiki/ASCII#Printable_character_table

Reply:
Oh, true. My bad 👍